src/Types/TypeRegistry.php

```diff
@@ -21,7 +21,10 @@ final class TypeRegistry
     /** @param array<string, Type> $instances */
     public function __construct(array $instances = [])
     {
-        $this->instances = $instances;
+        $this->instances = [];
+        foreach ($instances as $name => $type) {
+            $this->register($name, $type);
+        }
```

Comment:
While this is technically more correct than the previous behavior, I'm afraid that this change will significantly slow down the constructor because each `register()` call performs a lookup and as you already found out, that lookup is not as fast as it should be.

Reply:
We will have to mimick the `register()` function otherwise, what do you propose?

Reply:
once this bugfix PR is merged, https://github.com/doctrine/dbal/pull/6082 optimization PR is RTM and targetting the same branch, so it will be fast and O(1) (O(N) for whole constructor)

Reply:
I get that, but it's like you hand me the poison just to sell me the antidote afterwards. I cannot merge this change on its own. However, feel free to move the tests you've developed for this PR over to #6082.

Reply:
So you want me to close this PR in favor of #6082 incl. changes in this PR?

I made two PRs for better docs purposes, it should not be a problem, as this PR is functional on it's own, only the later PR will improve the performance.

Reply:
> So you want me to close this PR in favor of #6082 incl. changes in this PR?

That would be one way to do it, yes.

> I made two PRs for better docs purposes,

Which is a good idea in general.

> it should not be a problem, as this PR is functional on it's own,

No, it will decrease performance significantly which is why I cannot accept it in its current form.

Reply:
> That would be one way to do it, yes.

done, see #6082

Reply:
@derrabus may I ask you to review the #6082 PR, please?